nservicebus/nhibernate/custom-mappings.md

```diff
@@ -0,0 +1,126 @@
+
+
+## Custom .hbm.xml mapping
+
+Using NHibernate mapping files is the original way to customize your mapping. It needs to be created and then either embedded as a resource in your assembly or available on the file system.
+
+NOTE: Using xml mappings is the default approach for NHibernate.
+
+Look at these pages to see how you can create these mappings:
+
+* [Your first NHibernate based application](http://nhibernate.info/doc/tutorials/first-nh-app/your-first-nhibernate-based-application.html) (nhibernate.info)
+
+
+### Mappings as embedded resources
+
+It is not required to create your own NHibernate configuration object and pass it to the NServiceBus NHibernate configuration when using embedded resources. All other custom mapping options require the NHibernate configuration to be customized and passed to NServiceBus.
+
+
+### Read mappings from file system instead of embedded resources
+
+If you do not want to embed the `*.hbm.xml` files then you can also load mappings from the file system. Create a custom NHibernate configuration object and use the following example to add mappings from the file system.
+
+<!-- import NHibernateLoadMappingsFromFileSystem -->
+
+## Use Fluent NHibernate
+
+Fluent NHibernate gives you a type-safe mapping approach where the mapping is specified as c# code and not as xml but still seperate from the classes. The benefit is that you get compile time feedback when a mapping is not valid any more when you have breaking changes in your mapping but also that your classes and mappings are not part of the same .net type.
+
+To use it with NServiceBus:
+
+1. Install `FluentNHibernate` package via Nuget.
+2. Create a custom NHibernate configuration
+a. via FluentNHibernate
+b. or by creating a new Configuration instance and pass it to FluentNHibernate
+3. Pass it to the NServiceBus NHibernate configuration.
+
+
+Example of a possible implementation:
+
+<!-- import NHibernateInitWithFluentNHibernate -->
+
+References:
+
+* [Fluent NHibernate - Website](http://www.fluentnhibernate.org)
+* [Fluent NHibernate - Nuget package](http://www.nuget.org/packages/FluentNHibernate/)
+* [Fluent NHibernate - GitHub Repository](https://github.com/jagregory/fluent-nhibernate)
+* [Fluent NHibernate - Getting Started](https://github.com/jagregory/fluent-nhibernate/wiki/Getting-started)
+
+
+## Use NHibernate.Mapping.Attributes
+
+With NHibernate.Mapping.Attributes you can decorate your saga data classes. This keeps your classes, mapping and schema data very close. Your saga types have a dependency on the NHibernate.Mapping.Attributes assembly.
+
+How NHibernate.Mapping.Attributes works is that it needs to know which types it needs to scan, then it create a regular NHibernate mapping configuration that gets passed to your NHibernate configuration.
```

Comment:
`NHibernate.Mapping.Attributes` need to know which types to scan so to create a regular NHibernate mapping configuration that gets passed to your NHibernate configuration.